src/main/java/org/ehrbase/fhirbridge/ehr/converter/TherapyCompositionConverter.java

```diff
@@ -0,0 +1,264 @@
+
+    private static final Map<String, KategorieDefiningCode> kategorieMap = new HashMap<>();
+    private static final Map<String, NameDerProzedurDefiningCode> nameDerProzedurMap = new HashMap<>();
+    private static final Map<String, KoerperstelleDefiningCode> koerperstelleMap = new HashMap<>();
+    private static final Map<String, GeraetenameDefiningCode> geraetenameMap = new HashMap<>();
+
+    private static String SNOMED_SYSTEM = "http://snomed.info/sct";
+
+
+    static {
+        for (KategorieDefiningCode kategorie : KategorieDefiningCode.values()) {
+            kategorieMap.put(kategorie.getCode(), kategorie);
+        }
+
+        for (NameDerProzedurDefiningCode nameDerProzedurDefiningCode : NameDerProzedurDefiningCode.values()) {
+            nameDerProzedurMap.put(nameDerProzedurDefiningCode.getCode(), nameDerProzedurDefiningCode);
+        }
+
+        for (KoerperstelleDefiningCode koerperstelleDefiningCode : KoerperstelleDefiningCode.values()) {
+            koerperstelleMap.put(koerperstelleDefiningCode.getCode(), koerperstelleDefiningCode);
+        }
+
+        for (GeraetenameDefiningCode geraetenameDefiningCode : GeraetenameDefiningCode.values()) {
+            geraetenameMap.put(geraetenameDefiningCode.getCode(), geraetenameDefiningCode);
+        }
+    }
+
+
+    @Override
+    public Procedure fromComposition(GECCOProzedurComposition composition) throws CompositionConversionException {
+        // TODO: Implement
+        return null;
+    }
+
+    @Override
+    public GECCOProzedurComposition toComposition(Procedure procedure) throws CompositionConversionException {
+
+
+        if (procedure == null) {
+            return null;
+        }
+
+        GECCOProzedurComposition result = new GECCOProzedurComposition();
+
+        // set feeder audit
+        FeederAudit fa = CommonData.constructFeederAudit(procedure);
+        result.setFeederAudit(fa);
+
+        // Map Kategorie
+
+        result.setKategorie(new ArrayList<>());
+
+        for (Coding coding : procedure.getCategory().getCoding()) {
+            if (coding.getSystem().equals(SNOMED_SYSTEM) && kategorieMap.containsKey(coding.getCode())) {
+                GeccoProzedurKategorieElement element = new GeccoProzedurKategorieElement();
+                element.setValue(kategorieMap.get(coding.getCode()));
+
+                result.getKategorie().add(element);
+            }
+        }
+
+
+        // Map Start Time
+        try {
+
+            if (!procedure.getExtension().isEmpty() && procedure.getExtension().get(0).getValue() instanceof DateTimeType) {
+                result.setStartTimeValue(((DateTimeType) procedure.getExtension().get(0).getValue()).getValueAsCalendar().toZonedDateTime());
+            } else {
+                result.setStartTimeValue(procedure.getPerformedDateTimeType().getValueAsCalendar().toZonedDateTime());
+            }
+
+        } catch (NullPointerException | FHIRException ex) {
```

Comment:
Do you really want to catch here any FHIRException ?

Reply:
I will further specify this in my next commit